kernel/random/src/lib.rs

```diff
@@ -0,0 +1,88 @@
+//! A crate for generating random numbers.
+
+#![no_std]
+
+use rand_chacha::{
```

Comment:
do we need to use `rand_chacha`? I guess the real question is whether we need to provide a cryptographically-secure RNG as the default randomness provider in Theseus.... I don't think that we do.

Previously we were using `rand_core` via `rand` (with no features enabled); I think that's the way to go, and IIRC, is what `rand` recommends doing. We could also use other crates like `fastrand`, but considering that `wasmtime` uses `rand` directly, we might as well just stick with that.

We can also expose a crypto-secure RNG from this crate, or a separate crate, but I feel like that shouldn't be the only/default choice.

Reply:
My logic behind this was the user can seed whatever RNG they'd like with this global CSPRNG. Although I can see the benefit of having a global non-cryptographically secure PRNG so that crates don't have to depend on their own implementation (e.g. `fastrand`).